### PR TITLE
chore: upgrade react-native-reanimated to 2.7.0

### PR DESCRIPTION
**Pull request by khushal87**
## 🎯 Goal

The goal of this PR is to upgrade the dependency react-native-reanimated to 2.7.0. The decision was taken because there was an issue with the `__reanimatedWorkletInit` function and `__worklet` property while creating mocks of the package while testing.

## 🛠 Implementation details

Upgraded the package everywhere.

## 🎨 UI Changes

Nil

## ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [x] Documentation is updated
- [ ] New code is tested in main example apps, including all possible scenarios
  - [x] SampleApp iOS and Android
  - [x] Expo iOS and Android

### CODE REVIEW

File: docusaurus/docs/reactnative/basics/getting_started.mdx [hunk 1]
```diff
@@ -93,7 +93,7 @@ To be able to use the Stream Chat React Native SDK, a few dependencies must meet
 
 | react-native | stream-chat-react-native (min required) | react-native-reanimated (min required) |
 | ------------ | --------------------------------------- | -------------------------------------- |
-| 0.67         | 4.2.0                                   | 2.4.0                                  |
+| 0.67         | 4.2.0                                   | 2.7.0                                  |
```
Comment (santhoshvai) [inline docusaurus/docs/reactnative/basics/getting_started.mdx:96]: This is not true.. its more like only 4.11.0 will need 2.7.0

Comment (santhoshvai) [reply docusaurus/docs/reactnative/basics/getting_started.mdx:96]: also reanimated 2.6 adds support of RN 0.68, so lets add a new row and keep the previous one the same

0.68 - 4.11.0 - 2.7.0

I assume 4.11 gets triggered after we merge this

**Review by santhoshvai — CHANGES_REQUESTED** [review]
Just  requesting changes for the mdx file

File: docusaurus/docs/reactnative/basics/getting_started.mdx (hunk 1 above)
Comment (khushal87) [reply docusaurus/docs/reactnative/basics/getting_started.mdx:96]: Kinda confused if we should add for react native 0.68, but the other two make complete sense.

**Review by santhoshvai — APPROVED** [review]

**Pull request merged**